src/vm-memory/src/mmap.rs

```diff
@@ -51,9 +77,39 @@ impl GuestRegionMmap {
         Ok(GuestRegionMmap {
             mapping,
             guest_base,
+            dirty_bitmap: None,
+        })
+    }
+
+    /// Create a new memory-mapped memory region for the guest's physical memory.
+    /// This region will own a dedicated bitmap to handle dirty page tracking.
+    pub fn new_with_bitmap(
```

Comment:
Do we need a new function ? Wouldn't it be better to supply a flag to `new()` ?

Reply:
I will add a dedicated function to enable dirty page tracking instead and use the original constructor for both cases.